### PR TITLE
Defunct: Adding test to check interpolation methods.

### LINKED ISSUE #1038: Test each interpolation method against GDAL reprojection.
Test each resampling method for geotrellis.raster.reproject against GDAL reprojections.
@st028 is on this one.

### PR DESCRIPTION
**Pull request by SamThompson**
Will fix #1038

### CODE REVIEW

**Comment by st028** [comment]
As of now, this is the output for ReprojectSpec:


raster-test > test-only *ReprojectSpec
[info] ReprojectSpec:
[info] reprojects in approximation to GDAL
[info] - should (approximately) match a GDAL nearest neighbor interpolation on nlcd tile
[info] - should (approximately) match a GDAL bilinear interpolation on nlcd tile
[info] - should (approximately) match a GDAL cubic convolution interpolation on nlcd tile *** FAILED ***
[info]   Failed on (4, 2): 226.0 was not equal to 227.0 (ReprojectSpec.scala:102)
[info] - should (approximately) match a GDAL cubic spline interpolation on nlcd tile *** FAILED ***
[info]   Failed on (5, 0): 222.0 was not equal to 225.0 (ReprojectSpec.scala:132)
[info] - should (approximately) match a GDAL lanczos interpolation on nlcd tile *** FAILED ***
[info]   Failed on (5, 0): 222.0 was not equal to 225.0 (ReprojectSpec.scala:162)
[info] - should (approximately) match a GDAL average interpolation on nlcd tile *** FAILED ***
[info]   Failed on (0, 0): 211.0 was not equal to 226.0 (ReprojectSpec.scala:192)
[info] - should (approximately) match a GDAL mode interpolation on nlcd tile *** FAILED ***
[info]   Failed on (10, 0): 226.0 was not equal to 170.0 (ReprojectSpec.scala:222)
[info] - should (approximately) match a GDAL nearest neighbor on nlcd lat lon tile
[info] - should (approximately) match a GDAL bilinear on nlcd lat lon tile
[info] - should (approximately) match a GDAL cubic convolution interpolation on nlcd lat lon tile
[info] - should (approximately) match a GDAL cubic spline interpolation on nlcd lat lon tile *** FAILED ***
[info]   Failed on (5, 0): 226.0 was not equal to 225.0 (ReprojectSpec.scala:342)
[info] - should (approximately) match a GDAL lanczos interpolation on nlcd lat lon tile
[info] - should (approximately) match a GDAL average interpolation on nlcd lat lon tile *** FAILED ***
[info]   Failed on (5, 0): 226.0 was not equal to 224.0 (ReprojectSpec.scala:402)
[info] - should (approximately) match a GDAL mode interpolation on nlcd lat lon tile *** FAILED ***
[info]   Failed on (37, 0): 221.0 was not equal to 226.0 (ReprojectSpec.scala:432)
[info] - should (approximately) match a GDAL nearest neighbor on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:451)
[info] - should (approximately) match a GDAL bilinear on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:481)
[info] - should (approximately) match a GDAL cubic convolution interpolation on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:511)
[info] - should (approximately) match a GDAL cubic spline interpolation on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:541)
[info] - should (approximately) match a GDAL lanczos interpolation on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:571)
[info] - should (approximately) match a GDAL average interpolation on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:601)
[info] - should (approximately) match a GDAL mode interpolation on nlcd utm tile *** FAILED ***
[info]   562 was not equal to 578 (ReprojectSpec.scala:631)
[info] - should (approximately) match a GDAL nearest neighbor interpolation on slope tif and an error threshold of 0.125
[info] ScalaTest
[info] Run completed in 23 seconds, 937 milliseconds.
[info] Total number of tests run: 22
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 7, failed 15, canceled 0, ignored 0, pending 0
[info] *** 15 TESTS FAILED ***
[error] Failed: Total 22, Failed 15, Errors 0, Passed 7
[error] Failed tests:
[error] 	geotrellis.raster.reproject.ReprojectSpec
[error] (raster-test/test:testOnly) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 29 s, completed Mar 31, 2015 1:08:21 PM

File: vector/src/main/scala/geotrellis/vector/reproject/package.scala [hunk 1]
```diff
@@ -44,11 +44,11 @@ package object reproject {
       )
     }
 
-    def apply(extent: Extent, src: CRS, dest: CRS): Extent = {
+    def apply(extent: Extent, src: CRS, dest: CRS): Polygon = {
       val f = Transform(src, dest)
       val sw = f(extent.xmin, extent.ymin)
       val ne = f(extent.xmax, extent.ymax)
-      Extent(sw._1,sw._2,ne._1,ne._2)
+      Extent(sw._1,sw._2,ne._1,ne._2).toPolygon
```
Comment (lossyrob) [inline vector/src/main/scala/geotrellis/vector/reproject/package.scala]: This needs to not only be a Polygon, but needs to actually reproject each point of the polygon.

```
def apply(extent: Extent, src: CRS, dest: CRS): Polygon = 
  apply(extent.toPolygon, src, dest)
```

would work.

**Comment by lossyrob** [comment]
@moradology have we absorbed these test, so I can just close this?

**Comment by moradology** [comment]
Incorporating these tests could be done with a point or two of work, I ended up writing tests that verified functionality particular to the creation of boxes for aggregation and to the mean/median/mode resampled values.

Basically, a little bit of work wiring up the updated geotiff reader is all that would be required. I mostly fit the resampling code into the already working infrastructure, which is more what these test. Closing isn't a bad idea, I've got a copy in my local git and I'd be happy to update this and get it working.

**Comment by lossyrob** [comment]
Sounds like what these tests cover is already tested. I'm going to close this, thanks.

**Pull request closed without merging**